ntbs-service/Pages/ContactDetails/Index.cshtml.cs

```diff
@@ -17,25 +17,28 @@ public class IndexModel : PageModel
     {
         private readonly IUserService _userService;
         private readonly IReferenceDataRepository _referenceDataRepository;
+        private readonly IUserHelper _userHelper;
 
-        public IndexModel(IUserService userService, IReferenceDataRepository userRepository)
+        public IndexModel(IUserService userService, IReferenceDataRepository userRepository, IUserHelper userHelper)
         {
             _userService = userService;
             _referenceDataRepository = userRepository;
+            _userHelper = userHelper;
         }
 
         public User ContactDetails { get; set; }
         
         public IList<PHEC> RegionalMemberships { get; set; }
 
         [BindProperty(SupportsGet = true)]
-        public string Username { get; set; }
+        public int? UserId { get; set; }
 
         public async Task<IActionResult> OnGetAsync()
         {
-            ContactDetails = (Username == null)
-                ? await _userService.GetUser(User)
-                : await _referenceDataRepository.GetUserByUsernameAsync(Username);
+            var loggedInUser = await _userService.GetUser(User);
+            ContactDetails = UserId.HasValue
+                ? await _referenceDataRepository.GetUserByIdAsync(UserId.Value)
+                : loggedInUser;
```

Comment:
The way that you've rewritten this makes it slightly less efficient, as it will now fetch the logged in user regardless of whether or not it will use it. This is only a very minor concern though.